clang/lib/Format/TokenAnnotator.cpp

```diff
@@ -3497,6 +3497,15 @@ void TokenAnnotator::calculateFormattingInformation(AnnotatedLine &Line) const {
           Tok->setType(TT_TrailingReturnArrow);
           break;
         }
+        if (Tok->isNot(TT_TrailingAnnotation))
```

Comment:
Why do you skip here?

Reply:
Because `TT_TrailingAnnotation` is only the known case that needs special handling.

Reply:
But why? If there is no `Next` or it isn't a `l_paren` we also `continue`, shouldn't all that match to `TT_TrailingAnnotation`?

Reply:
Before: look for _any_ arrow after the `r_paren` of the function declaration.
Now: look for any arrow _except_ when inside the parentheses after a `TT_TrailingAnnotation`.

The case where `Next` is `nullptr` is handled by the condition of the `for` loop whereas any tokens (including non-`l_parens`) before a `TT_TrailingAnnotation` should be skipped as before.

Reply:
Understood.